services/blockchain-connector/shared/sdk/formatter.js

```diff
@@ -97,23 +97,27 @@ const formatBlock = (block) => {
 	const blockHeader = block.header;
 
 	const blockAssets = block.assets.map(asset => {
-		const assetModule = asset.module;
-		const blockAssetDataSchema = getBlockAssetDataSchemaByModule(assetModule);
-		const formattedAssetData = blockAssetDataSchema
-			? codec.decodeJSON(blockAssetDataSchema, Buffer.from(asset.data, 'hex'))
-			: asset.data;
-
-		if (!blockAssetDataSchema) {
-			// TODO: Remove this after all asset schemas are exposed (before tagging rc.0)
-			console.error(`Block asset schema missing for module ${assetModule}.`);
-			logger.error(`Unable to decode asset data. Block asset schema missing for module ${assetModule}.`);
-		}
+		// Decode asset data in case of binary payload
+		if (typeof (asset.data) === 'string') {
+			const assetModule = asset.module;
+			const blockAssetDataSchema = getBlockAssetDataSchemaByModule(assetModule);
+			const formattedAssetData = blockAssetDataSchema
+				? codec.decodeJSON(blockAssetDataSchema, Buffer.from(asset.data, 'hex'))
+				: asset.data;
+
+			if (!blockAssetDataSchema) {
+				// TODO: Remove this after all asset schemas are exposed (before tagging rc.0)
+				console.error(`Block asset schema missing for module ${assetModule}.`);
+				logger.error(`Unable to decode asset data. Block assnet schema missing for module ${assetModule}.`);
```

Comment:
Do we need both `console.error` and `logger.error`? Also maybe we can remove `TODO comment` as well since we are logging an error for safety.
```suggestion
				// TODO: Remove this after all asset schemas are exposed
				console.error(`Block asset schema missing for module ${assetModule}.`);
				logger.error(`Unable to decode asset data. Block asset schema missing for module ${assetModule}.`);
```